### PR TITLE
Allow for compilation in emscripten

### PR DESCRIPTION
**Pull request by timkpaine**
Related to https://github.com/apache/arrow/pull/3350

### CODE REVIEW

**Comment by timkpaine** [comment]
I signed it!

File: double-conversion/utils.h [hunk 1]
```diff
@@ -94,7 +94,8 @@ int main(int argc, char** argv) {
     defined(_MIPS_ARCH_MIPS32R2) || \
     defined(__AARCH64EL__) || defined(__aarch64__) || defined(__AARCH64EB__) || \
     defined(__riscv) || \
-    defined(__or1k__) || defined(__arc__)
+    defined(__or1k__) || defined(__arc__) \
```
Comment (paulnice) [inline double-conversion/utils.h]: should || be added before \?

Comment (timkpaine) [reply double-conversion/utils.h]: Yep

**Comment by floitsch** [comment]
Thanks, and sorry for the delay.

**Pull request merged**